src/trace/dma-trace.c

```diff
@@ -215,6 +215,14 @@ static int dma_trace_buffer_init(struct dma_trace_data *d)
 	return 0;
 }
 
+static void dma_trace_buffer_free(struct dma_trace_data *d)
+{
+	struct dma_trace_buf *buffer = &d->dmatb;
+
+	rfree(buffer->addr);
+	memset(buffer, 0, sizeof(*buffer));
```

Comment:
there is a spinlock on the init, is it needed when resetting the buffer?

````c
/* initialise the DMA buffer, whole sequence in section */
	spin_lock_irq(&d->lock, flags);

	buffer->addr  = buf;
	buffer->size = DMA_TRACE_LOCAL_SIZE;
	buffer->w_ptr = buffer->addr;
	buffer->r_ptr = buffer->addr;
	buffer->end_addr = (char *)buffer->addr + buffer->size;
	buffer->avail = 0;

	spin_unlock_irq(&d->lock, flags);
````

Reply:
probably a good thing to hold